src/btree/bt_cursor.c

```diff
@@ -79,6 +77,7 @@ __btcur_bounds_search_near_reposition(WT_SESSION_IMPL *session, WT_CURSOR_BTREE
 
     if (key_out_of_bounds) {
         __wt_cursor_set_raw_key(cursor, upper ? &cursor->upper_bound : &cursor->lower_bound);
+        *reposition_exactp = upper ? -1 : 1;
```

Comment:
We should leave a comment clarify why this is correct here.

Reply:
added a comment, let me know if you think it suffices

Reply:
I would update this comment to something along the lines of:
"
Given that we have positioned the cursor on either of the upper or lower bounds we can determine the "exact" return argument. If we positioned at the upper bound then any key we find must be less than or equal to that bound which is guaranteed to be less than the original search near key. The reverse holds true for the lower bound.
"